### PR TITLE
Add content-encoding header

### PR DESCRIPTION
**Pull request by paulius-valiunas**

### CODE REVIEW

File: tests/backend-storage/src/test/utils/Helpers.ts [hunk 1]
```diff
@@ -37,6 +38,14 @@ export async function queryAndAssertMetadata(
   expect(metadata).to.deep.equal(expectedMetadata);
 }
 
+export async function queryAndAssertHeaders(
```
Comment (austeja-bentley) [inline tests/backend-storage/src/test/utils/Helpers.ts:41]: the function seems to only assert content type. I think a better name would be `queryAndAssertContentType`

Comment (paulius-valiunas) [reply tests/backend-storage/src/test/utils/Helpers.ts:41]: contentEncoding is the only header I have implemented, my idea was to add every new header that's implemented. I couldn't find a way to get a list of Typescript interface members at run time.

Comment (paulius-valiunas) [reply tests/backend-storage/src/test/utils/Helpers.ts]: you're right, I renamed the helper

File: storage/azure/src/server/AzureServerStorage.ts [hunk 2]
```diff
@@ -212,6 +215,7 @@ export class AzureServerStorage extends ServerStorage {
       reference,
       size: contentLength!,
       metadata,
+      contentEncoding: _response.parsedHeaders.contentEncoding,
```
Comment (austeja-bentley) [inline storage/azure/src/server/AzureServerStorage.ts:217]: Are both `_response` and `parsedHeaders` always guaranteed to be not null? maybe it's worth to use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here

File: storage/azure/src/server/wrappers/BlockBlobClientWrapper.ts [hunk 3]
```diff
@@ -28,26 +30,53 @@ export class BlockBlobClientWrapper {
...
-    options?: MultipartUploadOptions
+    options?: MultipartUploadOptions,
+    headers?: ContentHeaders
   ): Promise<void> {
+    const blobHTTPHeaders: BlobHTTPHeaders | undefined =
+      headers === undefined
+        ? undefined
+        : {
+            blobContentEncoding: headers.contentEncoding,
+          };
```
Comment (austeja-bentley) [inline storage/azure/src/server/wrappers/BlockBlobClientWrapper.ts:67]: Duplicate with lines 38-43

File: storage/core/src/common/Interfaces.ts [hunk 4]
```diff
@@ -26,7 +28,7 @@ export interface ObjectReference extends ObjectDirectory {
   objectName: string;
 }
 
-export interface ObjectProperties {
+export type ObjectProperties = ContentHeaders & {
```
Comment (austeja-bentley) [inline storage/core/src/common/Interfaces.ts:33]: `contentEncoding` in `ContentHeaders` is nullable. Should it be nullable in `ObjectProperties`? I would assume that there is always some kind of default value specified or is this a wrong assumption?

**Review by austeja-bentley — COMMENTED** [review]
Added a few comments pointing out relatively minor details. Overall I like how this looks, thanks.

File: storage/core/src/common/Interfaces.ts (hunk 4 above)
Comment (paulius-valiunas) [reply storage/core/src/common/Interfaces.ts:33]: it's nullable in ContentHeaders, and it should be undefined by default

File: storage/azure/src/server/AzureServerStorage.ts (hunk 2 above)
Comment (paulius-valiunas) [reply storage/azure/src/server/AzureServerStorage.ts:217]: they are always defined, unless the type definitions are wrong

File: storage/azure/src/server/wrappers/BlockBlobClientWrapper.ts (hunk 3 above)
Comment (paulius-valiunas) [reply storage/azure/src/server/wrappers/BlockBlobClientWrapper.ts]: extracted into separate method

File: storage/core/src/common/Interfaces.ts (hunk 4 above)
Comment (paulius-valiunas) [reply storage/core/src/common/Interfaces.ts:33]: image
here's the definition of that property, it's nullable

Comment (austeja-bentley) [reply storage/core/src/common/Interfaces.ts:33]: Ah I see, ok

**Review by austeja-bentley — APPROVED** [review]

**Pull request merged**